Gateway/Request/TerminalIdDataBuilder.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Buckaroo\Magento2\Gateway\Request;
+
+use Buckaroo\Magento2\Logging\Log as BuckarooLog;
+use Magento\Framework\Stdlib\CookieManagerInterface;
+use Magento\Payment\Gateway\Request\BuilderInterface;
+
+class TerminalIdDataBuilder implements BuilderInterface
+{
+    private BuckarooLog $buckarooLog;
+    private CookieManagerInterface $cookieManager;
+
+    public function __construct(CookieManagerInterface $cookieManager, BuckarooLog $buckarooLog)
+    {
+        $this->cookieManager = $cookieManager;
+        $this->buckarooLog = $buckarooLog;
+    }
+
+    public function build(array $buildSubject): array
+    {
+        return ['terminalID' => $this->getPosPaymentTerminalId()];
+    }
+
+    /**
+     * @return null|string
+     */
+    private function getPosPaymentTerminalId(): ?string
+    {
+        $terminalId = $this->cookieManager->getCookie('Pos-Terminal-Id');
+        $this->buckarooLog->addDebug(__METHOD__ . '|1|');
```

Comment:
why do you need the logging here?

Reply:
I copied the functionality from here app/code/Buckaroo/Magento2/Model/Method/Pospayment.php:103. I don't know for what was used. I will remove it until we need it.